### PR TITLE
[RDY] Add introduction outlining the document.

### PR DESCRIPTION
**Pull request by gfokkema**

### CODE REVIEW

File: research/00-introduction.tex [hunk 1]
```diff
@@ -2,6 +2,30 @@ \section*{Introduction}
 \label{sec:introduction}
 \addcontentsline{toc}{section}{Introduction}
 
+Spoofax is a language workbench developed by TU Delft over the course of several
```
Comment (Hjdskes) [inline research/00-introduction.tex]: Delft University of Technology? Fancier :p

File: research/00-introduction.tex [hunk 2]
```diff
@@ -2,6 +2,30 @@ \section*{Introduction}
...
+years. During those years it has grown to `a platform for developing textual
```
Comment (Hjdskes) [inline research/00-introduction.tex]: If this is you paraphrasing someone, 1) use "", 2) italics and 3) most importantly, cite!

File: research/00-introduction.tex [hunk 3]
```diff
@@ -2,6 +2,30 @@ \section*{Introduction}
...
+domain-specific languages with full-featured Eclipse editor plugins'.
```
Comment (Hjdskes) [inline research/00-introduction.tex]: Also, IntelliJ is being worked towards, so perhaps this should be IDE-agnostic.

File: research/00-introduction.tex [hunk 4]
```diff
@@ -2,6 +2,30 @@ \section*{Introduction}
...
+Today languages in Spoofax are constructed using the meta-languages `SDF3',
+`NABL', `TS' and `Stratego', where each meta-language has a specific purpose.
+These meta-languages in turn are used to provide several language services,
```
Comment (Hjdskes) [inline research/00-introduction.tex]: `provide` -> `generate`?

File: research/00-introduction.tex [hunk 5]
```diff
@@ -2,6 +2,30 @@ \section*{Introduction}
...
+such as syntax highlighting, code transformation and error marking.
+
+Since Spoofax is used to develop new domain-specific languages, the ability
+for rapid prototyping using a read-eval-print loop would be very convenient.
+In this research report we first aim to get a better overview of the scope
```
Comment (Hjdskes) [inline research/00-introduction.tex]: No we in formal writing. Is it oke in an introduction though?

File: research/00-introduction.tex [hunk 6]
```diff
@@ -2,6 +2,30 @@ \section*{Introduction}
...
+REPLs and literate programming in order to clarify what functionality
+a good implementation should have.
+
+After having explored the domain of our project, we will give a concise
+problem definition. This problem definition will then be analyzed in the context
+of the services that Spoofax already provides to clarify potential problems
+and their solutions in creating a REPL that works generically across all
+generated languages.
+
+At last the requirements for the product will be analyzed, as well as the
```
Comment (Hjdskes) [inline research/00-introduction.tex]: `At last` == eindelijk. Perhaps you mean `Finally`, or `To conclude`?

File: research/00-introduction.tex [hunk 7]
```diff
@@ -2,6 +2,31 @@ \section*{Introduction}
...
 \addcontentsline{toc}{section}{Introduction}
 
+Spoofax is a language workbench developed by Delft University of Technology over
+the course of several years. During those years it has grown to \textit{"a
+platform for developing textual domain-specific languages with full-featured
+Eclipse editor plugins"}~\cite{SpoofaxWebsite}. Today languages in Spoofax are
+constructed using the meta-languages `SDF3', `NABL', `TS' and `Stratego', where
+each meta-language has a specific purpose. These meta-languages in turn are used
+to generate several language services, such as syntax highlighting, code
+transformation and error marking.
```
Comment (Balletie) [inline research/00-introduction.tex]: This repeats a lot of what is written in the introduction of the next section, not sure if that's a good idea.

File: research/00-introduction.tex [hunk 8]
```diff
@@ -2,6 +2,31 @@ \section*{Introduction}
...
+first aim of this research report is to get a better overview of the scope of
+Spoofax and each of its meta-languages. Afterwards, it will explore REPLs and
+literate programming in order to clarify what functionality a good
+implementation should have.
+
+After having explored the domain of our project, a concise problem definition
+will be given. This problem definition will then be analyzed in the context of
+the services that Spoofax already provides to clarify potential problems and
+their solutions in creating a REPL that works generically across all generated
+languages.
```
Comment (Balletie) [inline research/00-introduction.tex]: When phrased in this way it sounds as if we will not generate REPLs, but create one REPL to rule them all.

File: research/00-introduction.tex (hunk 7 above)
Comment (Hjdskes) [inline research/00-introduction.tex]: That's a good point. Perhaps we should cite section 1 :laughing: 

I think we can remove the last part:

> Today languages in Spoofax are constructed using the meta-languages `SDF3', `NABL', `TS' and `Stratego', where each meta-language has a specific purpose. These meta-languages in turn are used to generate several language services, such as syntax highlighting, code transformation and error marking.

File: research/00-introduction.tex [hunk 9]
```diff
@@ -2,6 +2,31 @@ \section*{Introduction}
...
+
+To conclude, the requirements for the product will be analyzed, as well as the
+methods that will be used in order to realize these requirements.
```
Comment (Balletie) [inline research/00-introduction.tex:24]: Maybe merge with previous paragraph? Not sure.. It's so small and lonely.

File: research/00-introduction.tex [hunk 10]
```diff
@@ -2,6 +2,31 @@ \section*{Introduction}
 \label{sec:introduction}
 \addcontentsline{toc}{section}{Introduction}
 
+Spoofax is a language workbench developed by Delft University of Technology over
+the course of several years. During those years it has grown to \textit{"a
+platform for developing textual domain-specific languages with full-featured
+Eclipse editor plugins"}~\cite{SpoofaxWebsite}. Today languages in Spoofax are
```
Comment (Balletie) [inline research/00-introduction.tex]: You can also cite the spoofax paper for this, as I have done

File: research/00-introduction.tex (hunk 7 above)
Comment (Balletie) [inline research/00-introduction.tex]: Yes I think we should do that.

**Pull request merged**